core/src/themes/ionic.mixins.scss

```diff
@@ -193,23 +193,33 @@
 
 @mixin rtl() {
   $root: #{&};
-  $selectors: str-split(#{add-root-selector($root, "[dir=rtl]")}, ",");
-  $dirSelector: "#{$root}:dir(rtl)";
+
+  $rootSplit: str-split($root, ",");
+  $selectorsSplit: str-split(#{add-root-selector($root, "[dir=rtl]")}, ",");
 
   // Selectors must be split into individual selectors in case the browser
   // doesn't support a specific selector.
-  // For example, Firefox and Safari doesn't support :host-context().
+  // For example, Firefox and Safari doesn't support `:host-context()`.
   // If an invalid selector is used, then the entire group of selectors
   // will be ignored.
   // @link https://www.w3.org/TR/selectors-3/#grouping
-  @each $selector in $selectors {
+  @each $selector in $selectorsSplit {
     @at-root #{$selector} {
       @content;
     }
   }
 
-  @at-root #{$dirSelector} {
-    @content;
+  // Adding :dir() in case the browser doesn't support `:host-context()` and does support `:dir()`.
+  // `:host-context()` is added:
+  // - through the `add-root-selector()` function.
+  // - first so that it takes precedence over `:dir()`.
+  // For example, Firefox doesn't support `:host-context()`, but does support `:dir()`.
+  // MDN says that `:dir()` is supported by Safari, but it doesn't seem to work.
```

Comment:
This feature was only supported starting in iOS 16.4: https://webkit.org/blog/13966/webkit-features-in-safari-16-4/

This may be why it did not work. Can we update the comment?

Reply:
Actually question regarding this: Is `:dir` required for this fix to work? Last time I reviewed the change the `:dir` usage was not present.

Reply:
Did some digging and the answer is yes. In that case, can we clarify the comment as per above?

Reply:
So Firefox/Safari will still work as expected if we remove `:dir`?

Reply:
Yes, the only time that `:dir` shines is with a nested popover. The arrow doesn't rotate correctly on Safari and Firefox b/c they don't handle `:host-context`. With `:dir`, the arrow renders correctly on Firefox.
Honestly, with no major changes, the addition of `:dir` is a nice to have as of now.

Reply:
Sounds good. Let's keep it as is since more browsers are starting to adopt `:dir`.

Reply:
Awesome, then it's ready for you to review.